### DISCUSSION TITLE
Timeout rule.

### DISCUSSION THREAD

**Discussion by web5173** [pocketbase/pocketbase#5839] [Q&A]
How to set an API rule that does not allow modification after exceeding a specified time?

**Comment by ganigeorgiev** [comment]
Please try to be more specific. It depends on whether the time limit is fixed, or if not - who should be able to set/modify it. Without details about the actual use case and the desired behavior it is really difficult to give a proper answer or recommendation.

In general to target the _current_ time you may want to use one of the [datetime macros](https://pocketbase.io/docs/api-rules-and-filters/#-macros). Assuming that you want to target an individual time limit that could differ for each record from your collection, then in this case you can add a `date` type field (e.g. "endDate") to your collection and set the Update API rule to something like:
```js
@now <= endDate &&
// optional prevent regular users from changing the field
// if they shouldn't be allowed to do that
@request.data.endDate:isset = false
```

If the above doesn't help, please try to elaborate a little more on your use case.

**Comment by web5173** [comment]
Very good

**Comment by Erik0806** [comment]
Hi, is there a way to allow modifications only for eg. ten minutes after the creation? I tried to do that with operands and calculating with the created and @now but that doesn't seem to work. Can you give me a headsup here? Thanks for PocketBase!

**Reply by ganigeorgiev** [reply]
Math expressions are not supported in the API rules if that's what you tried.

For more advanced access controls you'll have to use the Go or JS app event hooks:
- https://pocketbase.io/docs/go-event-hooks/
- https://pocketbase.io/docs/js-event-hooks/

**Reply by Erik0806** [reply]
ok, thanks for the fast response. Yeah I tried with math expessions. Are there plans to add them to the api rules in the future? That could be really helpful. Have a nice day :)
